board.h

```diff
@@ -0,0 +1,73 @@
+#pragma once
+
+#include <iostream>
+#include <algorithm>
+#include <vector>
+#include "pieces.h"
+#include "constants.h"
+
+struct indices {
+    int i, j;
+};
+
+template <typename T>
+Piece* Create() { return new T(); }
+
+
+typedef Piece* (*CreateFn)();
+
+
+class Board {
+public:
+    // playfield is 10 wide 40 tall.
+    // cell called using y * 10 + x
+    std::vector<int> board;
+    std::vector<int> blocksPerRow;
+    static const std::vector<CreateFn> pieceNames;
+    Piece* currentPiece;
+    int absPiecePositionRow;
+    int absPiecePositionCol;
+    int rotation;
+    int currentHeight;
+
+
+    Board();
+
+    int& operator[](int rowCol);
+
+    int& operator[](indices rowCol); // called with board({row, col})
```

Comment:
why not just return a pair<int,int>?

Reply:
Why would it need to? It returns the value in that cell.